azurerm/internal/services/frontdoor/frontdoor_custom_https_configuration_resource.go

```diff
@@ -0,0 +1,312 @@
+package frontdoor
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/frontdoor/mgmt/2020-01-01/frontdoor"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/frontdoor/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmFrontDoorCustomHttpsConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmFrontDoorCustomHttpsConfigurationCreateUpdate,
+		Read:   resourceArmFrontDoorCustomHttpsConfigurationRead,
+		Update: resourceArmFrontDoorCustomHttpsConfigurationCreateUpdate,
+		Delete: resourceArmFrontDoorCustomHttpsConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(6 * time.Hour),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(6 * time.Hour),
+			Delete: schema.DefaultTimeout(6 * time.Hour),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"front_door_name": {
```

Comment:
I think the best solution is to actually add a `frontend_endpoint_id` and remove both `front_door_name` and `frontend_endpoint_name` as they can be derived from the id. That way it should be easier to manage from a user perspective and the implementation should not be more complex after removing those 2 fields. How does such a solution sound to you?

Reply:
That's a good idea, but because of the way Terraform works the core would not be able to build a proper dependency graph during the plan phase, if I am not mistaken, which is why we generally use the parent resources id attribute. You can give it a try and see if it builds a correct provisioning graph, but my gut is telling me it won't be able to connect the two together.

Reply:
In this case the parent resource for the custom https configuration is the frontend endpoint. I've started work on this and it looks good so far. Hopefully won't be long before I commit it.